### PR TITLE
adding version checks for backends

### PR DESCRIPTION
**Pull request by scarrazza**
Implements a simple version check for backends, including the mirroring with setup.py.

### CODE REVIEW

File: src/qibo/backends/__init__.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 from qibo import config
 from qibo.config import raise_error, log, warnings
 
+# versions requirements
+TF_MIN_VERSION = '2.2.0'
```
Comment (stavros11) [inline src/qibo/backends/__init__.py:6]: We could also move this to config.py.

**Review by stavros11 — COMMENTED** [review]
Thanks, the check looks good but only checks the tensorflow version. If I understood correctly your comment [here](https://github.com/Quantum-TII/qibo/pull/411#issuecomment-841206778) we would also like to check that the qibotf version is compatible with the qibo version.

**Comment by scarrazza** [comment]
I believe the only strong requirement we have now is the tf version, given that for qibotf we don't have any specific API that requires locking their version.

File: src/qibo/backends/__init__.py (hunk 1 above)
Comment (scarrazza) [reply src/qibo/backends/__init__.py:6]: Any special reason to move that to `config.py`?

Comment (stavros11) [reply src/qibo/backends/__init__.py:6]: No, other than that we define all other "constants" used throughout the qibo code there. I don't have any strong preference on what is the right position for `TF_MIN_VERSION` though.

**Comment by scarrazza** [comment]
@stavros11, yes, thanks, I agree.

**Pull request merged**